modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/vm/TcpDiscoveryVmIpFinder.java

```diff
@@ -235,15 +246,28 @@ else if (ipStr.indexOf(':') >= 0)
 
                     return res;
                 }
-                catch (IllegalArgumentException | UnknownHostException e) {
+                catch (IllegalArgumentException e) {
                     throw new IgniteSpiException(errMsg, e);
                 }
             }
             else {
                 try {
                     int port = Integer.parseInt(portStr);
 
-                    return Collections.singleton(new InetSocketAddress(addrStr, port));
+                    Collection<InetSocketAddress> col = new LinkedHashSet<>();
```

Comment:
I wonder if we really need this if-else here. Can we have same logic block for range and single port?

Reply:
The code was updated. Please check one more time.